ocw_data_parser/ocw_data_parser_test.py

```diff
@@ -19,12 +161,140 @@ def test_no_params(self):
         with self.assertRaises(Exception):
             OCWParser()
 
-    def test_ocw_parser_output(self):
+    def test_html_parser_output_list(self):
         """
-        Test that OCWParser handles downloaded json
+        Test passing in an output_list to the CustomHTMLParser
         """
-        ocw_parser = OCWParser(course_dir="ocw_data_parser/test_json/course_dir",
-                               destination_dir="ocw_data_parser/test_json/destination_dir")
-        with open('ocw_data_parser/test_json/master_json.json', 'r') as master_json:
-            master_json_data = json.loads(master_json.read())
-            self.assertEqual(master_json_data, ocw_parser.master_json)
+        output_list = ['test']
+        parser = CustomHTMLParser(output_list=output_list)
+        self.assertEqual('test', parser.output_list[0])
+
+    def test_parser_loaded_jsons(self):
+        """
+        Test instantiating a parser with preloaded jsons
+        """
+        self.set_up()
+        try:
+            OCWParser(loaded_jsons=self.ocw_parser.jsons)
+            self.tear_down()
+        except:
+            self.tear_down()
+            self.fail("instantiating parser with preloaded jsons failed")
+
+    def test_generate_master_json_none_source(self):
+        """
+        Make sure that running generate_master_json doesn't throw an error after nulling 
+        out the parser's source jsons
+        """
+        self.set_up()
+        self.ocw_parser.jsons = None
+        self.ocw_parser.generate_master_json()
+        self.assertIsNotNone(self.ocw_parser.jsons)
+        self.tear_down()
+
+    def test_generate_master_json_none_course_image_uid(self):
+        """
+        Make sure course_image_uid is regenerated by generate_master_json after it's nulled out
+        """
+        self.set_up()
+        self.ocw_parser.course_image_uid = None
+        self.ocw_parser.generate_master_json()
+        self.assertIsNotNone(self.ocw_parser.course_image_uid)
+        self.tear_down()
+
+    def test_load_raw_jsons_invalid_file(self):
+        """
+        Add a json file with invalid content to the course_dir and make sure it generates an exception
+        """
+        with open("ocw_data_parser/test_json/course_dir/jsons/test.json", "w") as f:
+            f.write("")
+        with self.assertRaises(Exception):
+            self.set_up(generate_static_site=True)
+        os.remove("ocw_data_parser/test_json/course_dir/jsons/test.json")
+        self.tear_down()
+
+    @mock_s3
```

Comment:
Is there a test where we don't want to use `mock_s3`? You can create an autouse fixture to make it apply to all tests in this file automatically:

    @pytest.fixture(autouse=True)
    def apply_mock_s3():
        with mock_s3():
            yield

    def test_upload_all_data_to_s3():
        ...

With `autouse` fixtures you don't need to specify it in the arguments of the test function for it to take effect.